internal/providers/system/system.go

```diff
@@ -56,3 +71,31 @@ func fetchConfig(logger *log.Logger, filename string) (types.Config, report.Repo
 	}
 	return util.ParseConfig(logger, rawConfig)
 }
+
+// fetchBaseConfig is a helper function to merge all the base config fragments inside of a particular directory.
+func fetchBaseConfig(logger *log.Logger, dir string) (types.Config, report.Report, error) {
+	var baseConfig types.Config
+	var ret report.Report
+	path := filepath.Join(distro.SystemConfigDir(), dir)
+	configs, err := ioutil.ReadDir(path)
+	if os.IsNotExist(err) {
+		logger.Info("no config dir at %q", path)
+		return types.Config{}, report.Report{}, nil
+	} else if err != nil {
+		logger.Err("couldn't read config dir %q: %v", path, err)
+		return types.Config{}, report.Report{}, err
+	}
+	if len(configs) == 0 {
+		logger.Info("no configs at %q", path)
+		return types.Config{}, report.Report{}, nil
+	}
+	for _, config := range configs {
+		intermediateConfig, intermediateReport, err := fetchConfig(logger, filepath.Join(path, config.Name()))
+		if err != nil {
+			return types.Config{}, report.Report{}, err
```

Comment:
The corresponding report may be non-empty.  We should capture that in the return.  We could merge it into the base report first but I don't think that's necessary.

```suggestion
			return types.Config{}, intermediateReport, err
```